fq.go

```diff
@@ -0,0 +1,170 @@
+package fq
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service time
+type FQScheduler struct {
+	lock         sync.Mutex
+	queues       []*Queue
+	clock        clock.Clock
+	vt           float64
+	C            float64
```

Comment:
`C` is naturally an integer.  I suspect we may have fewer typecasts if we give `C` an integer type.

Reply:
updated